### ISSUE TITLE
Predbat-led car charging to 100% stops before car is completely full

### ISSUE THREAD

**Issue by mpartington** [springfall2008/batpred#884]
**Describe the bug**
Have recently switched to Agile and am now using Predbat to manage my car charging slots, which is working brilliantly! I have linked the SoC from my car, so use this to tell Predbat the target SoC.

What I have found is if 100% is selected, then Predbat immediately stops the charge soon as it hits it.  If I was on say a GO slot, then the car would continue to charge for a while longer after it hits 100%. The car BMS would gradually taper the charge and perform cell balancing during this time.

Something else I noticed is that Predbat turned off the charging slot at 11.38 (ish). Given the slots chosen were consecutive, it's odd it didn't just charge through.

**Expected behaviour**
Rather than set "binary_sensor.predbat_car_charging_slot" to false soon as the car reports 100% SoC, allow the car to carry on charging until it's BMS stops the charge.  This would require either:
  - Predbat to monitor an active power entity for the charger and wait until it is zero (and soc =100)
  - Predbat to allow the car to charge to the end of the slot if 100% target selected -*** not 100% fix, but maybe easiest to implement ****
  - Predbat to add an additional margin to the kWh required (may cause an extra slot to be generated)

I realise another way would be to use the "car charging manual SoC" and set it to a higher than required kWh to force this.
 But that's a bit of a hack

**Predbat version**

V7.16.8

**Screenshots**
I ran out of time in this example (**normal GO slot**) , so the charger shuts of at 4.30. However you can see it charging beyond 100%, it only took on another 0.5kWh, so probably less than 1kWh had it had more time.  The 'lost' charge isn't really an issue, it's just giving the cells chance to balance.

![Screenshot_2024-03-24-17-46-50-755_io homeassistant companion android](https://github.com/springfall2008/batpred/assets/106986397/28653e40-a427-4799-893a-e340c0d1f6b5)

This is the recent Predbat led charge schedule (expected taper not to scale)
<img width="512" alt="image" src="https://github.com/springfall2008/batpred/assets/106986397/538795c2-f029-4ef6-8233-0657b1471675">

**Comment by mpartington** [comment]
Probably just me this is an issue for, and I've managed to fix locally.. So propose to close.

I've created a template sensor to calculate soc_kWh from soc%. I then use this via an automation to set "Car manual SOC kWh" minus a configurable amount once the charger is plugged in. This is then locked in and Predbat calculates the required slots based on my Mac charge setting.

I notice that the Renault app is a bit flaky, and sometimes soc doesn't come through for a period, which could mean (I assume) Predbat wouldn't see a soc rise and keep setting new (higher priced) slots.  Hopefully this method also mitigates against that failure mode.
